src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -207,36 +207,113 @@ bool ngraph::snippets::pass::PropagatePrecision::validate_and_infer_types_and_re
         op->validate_and_infer_types();
     }
 
+    // one flag for both options: only one result consumer
+    bool result_consumer_was_handled = false;
     for (size_t i = 0; i < op->get_output_size(); ++i) {
         auto output = op->output(i);
 
         if (output.get_element_type() != op_output_types[i]) {
             was_updated = true;
+            auto inputs_to_handle = output.get_target_inputs();
+            // if at least consumer is result then we should add conversion
+            // in this case we can not remove existing conversion with result consumers
+            const auto least_consumer_is_result = std::any_of(
+                inputs_to_handle.begin(),
+                inputs_to_handle.end(),
+                [](const Input<Node>& input) { return ngraph::is_type<ngraph::op::Result>(input.get_node()); });
```

Comment:
Seems like we can create lambda function since we call this check several times:
```
auto is_consumer_result = [](const std::set<ov::Input<ov::Node>> consumers) {
    return  std::any_of(inputs_to_handle.begin(), inputs_to_handle.end(),
                [](const Input<Node>& input) { return ngraph::is_type<ngraph::op::Result>(input.get_node()); });
};
```